template/.github/workflows/build.yml.j2

```
@@ -411,10 +417,16 @@ jobs:
         with:
           crate: cargo-edit
           bin: cargo-set-version
-      - name: Update version if PR
-        # For PRs to be merged against a release branch, the version has already been set, in which case ignore this step.
+      - name: Update version if PR against main branch
         if: ${{ github.event_name == 'pull_request' && github.ref == 'refs/heads/main' }}
         run: cargo set-version --offline --workspace 0.0.0-pr${{ github.event.pull_request.number }}
+      - name: Update version if PR against non-main branch
+        # For PRs to be merged against a release branch, use the version that has already been set in the calling script.
+        if: ${{ github.event_name == 'pull_request' && github.ref != 'refs/heads/main' }}
+        run: |
+          MANIFEST_VERSION=$(cargo metadata --format-version 1 --no-deps | jq -r '.packages[0].version')
+          PR_VERSION="${MANIFEST_VERSION}-pr${{ github.event.pull_request.number }}"
+          cargo set-version --offline --workspace "$PR_VERSION"
```

Ideally we shouldn't use github interpolation in scripts, but rather create env vars for the step/job/workflow, and use those in the script.

```suggestion
      - name: Update version if PR against non-main branch
        # For PRs to be merged against a release branch, use the version that has already been set in the calling script.
        if: ${{ github.event_name == 'pull_request' && github.ref != 'refs/heads/main' }}
        env:
          PR_NUMBER: ${{ github.event.pull_request.number }}
        run: |
          MANIFEST_VERSION=$(cargo metadata --format-version 1 --no-deps | jq -r '.packages[0].version')
          PR_VERSION="${MANIFEST_VERSION}-pr${PR_NUMBER}"
          cargo set-version --offline --workspace "$PR_VERSION"
```

In this case, it is ok, since the value can only be a number, and no injection can occur, and the workflow will be restructured to work more like the docker-images workflows.

---

I'll make another suggestion for the other occurrence. Feel free to ignore the suggestions as we can clean it up during the restructure.